src/sonic-ctrmgrd/tests/container_startup_test.py

```diff
@@ -0,0 +1,391 @@
+            common_test.STATE_DB_NO: {
+                common_test.FEATURE_TABLE: {
+                    "snmp": {
+                        "container_id": "any",
+                        "container_version": "",
+                        "current_owner": "any",
+                        "remote_state": "ready",
+                        "system_state": "up"
+                    }
+                },
+                common_test.KUBE_LABEL_TABLE: {
+                    "SET": {
+                        "snmp_enabled": "no_change"
+                    }
+                }
+            }
+        },
+        common_test.POST: {
+            common_test.STATE_DB_NO: {
+                common_test.FEATURE_TABLE: {
+                    "snmp": {
+                        "container_version": "20201230.11",
+                        "current_owner": "kube",
+                        "remote_state": "running",
+                        "system_state": "up"
+                    }
+                },
+                common_test.KUBE_LABEL_TABLE: {
+                    "SET": {
+                        "snmp_enabled": "no_change"
+                    }
+                }
+            }
+        }
+    },
+    7: {
+        common_test.DESCR: "kube container starting with this version blocked",
+        common_test.ARGS: "container_startup -f snmp -o kube -v 20201230.11",
+        common_test.PRE: {
+            common_test.CONFIG_DB_NO: {
+                common_test.FEATURE_TABLE: {
+                    "snmp": {
+                        "set_owner": "kube"
+                    }
+                }
+            },
+            common_test.STATE_DB_NO: {
+                common_test.FEATURE_TABLE: {
+                    "snmp": {
+                        "container_id": "no change",
+                        "container_version": "no change",
+                        "current_owner": "no change",
+                        "remote_state": "no change",
+                        "system_state": "up"
+                    }
+                },
+                common_test.KUBE_LABEL_TABLE: {
+                    "SET": {
+                        "snmp_20201230.11_enabled": "false"
+                    }
+                }
+            }
+        },
+        common_test.POST: {
+            common_test.STATE_DB_NO: {
+                common_test.FEATURE_TABLE: {
+                    "snmp": {
+                        "container_id": "no change",
+                        "container_version": "no change",
+                        "current_owner": "no change",
+                        "remote_state": "no change",
+                        "system_state": "up"
+                    }
+                },
+                common_test.KUBE_LABEL_TABLE: {
+                    "SET": {
+                        "snmp_20201230.11_enabled": "false"
+                    }
+                }
+            }
+        }
+    }
+}
+
+
+class TestContainerStartup(unittest.TestCase):
```

Comment:
Why use `unittest.TestCase`? You're already installing pytest via `tests_require=` of setup.py. Prefer pytest over unittest per the rest of the SONiC codebase.

Reply:
Dropped "import unittest" and only import TestCase as supported here.

This is what I observe python3 packages that uses class object in TestCode.
Please let me know, if you would like any further update. A reference would be a big help.